### DISCUSSION TITLE
MantineProvider button not having styles during loading

### DISCUSSION THREAD

**Discussion by dexter4life** [mantinedev/mantine#3492] [Q&A]
![image](https://user-images.githubusercontent.com/10168414/216735656-efd114aa-6f67-474a-ac54-cd7c1a469171.png)

during loading the buttons will be like that and after loading is complete then the style will be back. 

How do i fix this?

**Comment by rtivital** [comment]
You have not configured ssr logic. Use one of the templates as reference.

**Reply by dexter4life** [reply]
Can you show me an example please?

**Reply by rtivital** [reply]
Select template of your stack – https://github.com/mantinedev

**Reply by dexter4life** [reply]
The template idea is nice, and I am using it now. Although I am still getting that problem. See if I leave it as it is, then the CSS class will be `mantine-` and everything will be fine. But if I set it anything other than mantine, it will then give me that as seen on the screenshot. Please any suggestions?

**Reply by rtivital** [reply]
I suggest you to use a template. Users do not care about class names.

**Reply by dexter4life** [reply]
That's what the company wants. So what can I do?

**Reply by rtivital** [reply]
https://mantine.dev/theming/emotion-cache/#server-side-rendering-with-custom-cache

**Reply by dexter4life** [reply]
Is affecting a lot of things. For example, when the user logins, the whole page will be like there's no style on it for a while. I have to use preloader to cover it before a full load is completed.

**Reply by dexter4life** [reply]
thank a lot is working now. Lovely

**Reply by dexter4life** [reply]
Hello, Vitaly I notice that the method didn't work for nextjs with expressjs backend. What can I do to make it work?

**Reply by rtivital** [reply]
Use one of the templates as reference. If you have a custom setup, it is up to you to figure out how to setup ssr with it. You can browser source code of `@mantine/ssr` and `@mantine/next` packages to learn more.
